tests/perf.js

```diff
@@ -103,33 +113,126 @@ async function testPerf(runs, batches, testName) {
   console.log(`runs: ${runs} batches: ${batches} total: ${runs * batches}`);
   switch (testName) {
     case 'promise':
-      console.log('testing promise');
+      console.log('testing promise JSON read');
       printPerf(
-        'stream',
-        await run([], 0, () => {
+        testName,
+        await run([], 0, async () => {
```

Comment:
I don't really have a problem with changing these to async as it ultimately does not impact anything. However, I hope you understand that a function does not need to be async to return a promise.